src/devices/CharacterLcd/ICharacterLcd.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Iot.Device.CharacterLcd
+{
+    /// <summary>
+    /// Interface for character LCD Low-Level handler
+    /// </summary>
+    public interface ICharacterLcd : IDisposable
```

Comment:
If we only have one CharacterLcd, it might not be worth it to add an interface abstraction just yet, but instead wait till we have more. I specifically think of this because interfaces don't version very well in case we need to add stuff to the abstraction later

Reply:
There's currently only one implementation, but users are free to implement their own. (I.e it will be rather simple to also implement this interface for 7-Segment or 16-segment displays). Also, the only current implementation is named Hd44780, not a very intuitive name to use as handle name in other places. 
Then, this is used by LcdConsole, which is independent of the actual implementation of the Hd44780 and should not be hardlinked to that. 
Therefore I do think that the interface is the right choice here.

Reply:
Is multi-inheritance a concern for implementations of this interface like Hd44780? In case it isn't, I would suggest to instead use abstract classes so that you can version them better. I know that the new csharp features include interface default implementations, but I'm not sure if those work everywhere we want to run, since we currently target netcoreapp2.1

Reply:
In the case we have so far, a base class would do, but I can imagine cases where this would be a limitation, like when you derive from some binding to add the capability of the `ICharacterLcd`. There might be devices where it doesn't make sense to support the capabilities in the base implementation, but in derived classes only. 
I actually don't quite understand why you think that interfaces don't version well. Can you explain that?

Reply:
> Can you explain that?

Sure, what I mean is that (at least before C# added the feature for default implementations) once you ship an interface you should never add/remove or modify members in the future because this would be a big breaking change for consumers that have already implemented the interface or taken a dependency on one of the members. Because of this, we are very careful when introducing interfaces on our API, as you basically need to make sure that you get the API shape right the first time you ship it and won't want to add anything in the future. This is why if you actually check in System.Device.Gpio, we actually don't have that many interfaces, and instead use abstract classes for polymorphism.

Anyways, for this interface in particular I'm probably not super worried, because I expect most of the classes that implement it to to actually live on this project as well, which means that taking an update on one should update the implementing classes as well, that is why I asked if multi-inheritance was a concern, if it isn't then just use abstract classes and won't have this problem, but if you are concerned about it (which seems like you are) we can keep it an interface and document the breaking change.

Reply:
There's no binding for a 7-segment or a 16-segment display right now (I think), otherwise I could test whether the interface works on those, at least conceptionally. So we would know that it should hold at least for its basic purpose.

Reply:
@joperezr : After addressing https://github.com/dotnet/iot/pull/859#discussion_r394592045, I think it's still the most flexible approach to leave this as interface. The interface has basically the same methods as the Hd44780 implementation had previously, so I guess it should be rather stable. Or do you think there's anything missing or not generic enough? 
Otherwise, this would be ready for merge.